### DISCUSSION TITLE
Link metadata retrieval

### DISCUSSION THREAD

**Discussion by erlend-sh** [raffomania/linkblocks#54] [Ideas]
When a link is added, meta-data could be automatically pulled for it with something like [link-preview](https://github.com/EstebanBorai/link-preview) or [embed-service](https://github.com/Lantern-chat/embed-service) (they do slightly different things, but the exact difference is unclear to me).

> Retrieve website metadata such as
> - title
> - description
> - preview image
> - author
> - and more
> 
> from OpenGraph, Google, Schema.org and Twitter compliant sites

_Description_ and _preview image_ should probably be reserved for an optional “rich view” of some sort, but _title_ and _author_ would be great to include by default.

**Comment by kimlimjustin** [comment]
I just went through the `link-preview`, but after digging into the source code, I found out that unfortunately `author` wasn't actually provided by it. Instead, it only provides `title`, `description`, `domain`, and `image_url`. 

Should we include `title` by default with optional "rich view" (`image` and `description`) that is defaulted to be turned off but could be turned on for every list?

**Reply by erlend-sh** [reply]
Sounds good to me. What do you think @raffomania ?

**Reply by kimlimjustin** [reply]
Opened a PR for `title` enabled by default (#63), please let me know if there any further improvement needed
